nemo_aligner/algorithms/ppo.py

```diff
@@ -348,9 +469,11 @@ def run_training(self, dataloader_iter):
         return loss_mean, metrics
 
     def fit(self):
-        if (not isinstance(self.train_dataloader.batch_sampler, MegatronPretrainingRandomSampler)) and (
-            self.cfg.max_epochs is not None and self.cfg.max_epochs > 1
-        ):
+        if (
+            not isinstance(
+                self.train_dataloader_builder(consumed_samples=0).batch_sampler, MegatronPretrainingRandomSampler
```

Comment:
Can we move this check to `__init__()` where we already create a `train_dataloader`? (just to cleanup `fit()` a bit)

Reply:
3d6504441e0023d370373a99b1ba593783ee5406